cpp/src/rolling/detail/range_utils.cuh

```diff
@@ -0,0 +1,869 @@
+                 cudf::data_type_error);
+    return window_bounds<OrderbyT, ScalarT>(
+      orderby, grouping, nulls_at_start, row_delta, stream, mr);
+  }
+
+  template <typename OrderbyT, CUDF_ENABLE_IF(cudf::is_fixed_point<OrderbyT>())>
+  [[nodiscard]] std::unique_ptr<column> operator()(
+    column_view const& orderby,
+    std::optional<preprocessed_group_info> const& grouping,
+    bool nulls_at_start,
+    scalar const* row_delta,
+    rmm::cuda_stream_view stream,
+    rmm::device_async_resource_ref mr) const
+  {
+    CUDF_EXPECTS(!row_delta || (orderby.type().id() == row_delta->type().id()),
+                 "Orderby column and row_delta must both be fixed point.",
+                 cudf::data_type_error);
+    // TODO: Push this requirement onto the caller and just check for
+    // equal scales (avoids a kernel launch to rescale)
+    CUDF_EXPECTS(!row_delta || row_delta->type().scale() >= orderby.type().scale(),
+                 "row_delta must have at least as much scale as orderby column.",
+                 cudf::data_type_error);
+    if (row_delta && row_delta->type().scale() != orderby.type().scale()) {
+      auto const value =
+        dynamic_cast<fixed_point_scalar<OrderbyT> const*>(row_delta)->fixed_point_value(stream);
+      auto const new_scalar = cudf::fixed_point_scalar<OrderbyT>{
+        value.rescaled(numeric::scale_type{orderby.type().scale()}), true, stream};
+      return window_bounds<OrderbyT>(orderby, grouping, nulls_at_start, &new_scalar, stream, mr);
+    }
+    return window_bounds<OrderbyT>(orderby, grouping, nulls_at_start, row_delta, stream, mr);
+  }
+
+  template <typename OrderbyT, CUDF_ENABLE_IF(cudf::is_numeric_not_bool<OrderbyT>())>
+  [[nodiscard]] std::unique_ptr<column> operator()(
+    column_view const& orderby,
+    std::optional<preprocessed_group_info> const& grouping,
+    bool nulls_at_start,
+    scalar const* row_delta,
+    rmm::cuda_stream_view stream,
+    rmm::device_async_resource_ref mr) const
+  {
+    CUDF_EXPECTS(!row_delta || cudf::have_same_types(orderby, *row_delta),
+                 "Orderby column and row_delta must have the same type.",
+                 cudf::data_type_error);
+    return window_bounds<OrderbyT>(orderby, grouping, nulls_at_start, row_delta, stream, mr);
+  }
+
+  template <typename OrderbyT,
+            CUDF_ENABLE_IF(cuda::std::is_same_v<OrderbyT, cudf::string_view> &&
+                           (WindowTag == window_tag::CURRENT_ROW ||
+                            WindowTag == window_tag::UNBOUNDED))>
+  [[nodiscard]] std::unique_ptr<column> operator()(
+    column_view const& orderby,
+    std::optional<preprocessed_group_info> const& grouping,
+    bool nulls_at_start,
+    scalar const* row_delta,
+    rmm::cuda_stream_view stream,
+    rmm::device_async_resource_ref mr) const
+  {
+    CUDF_EXPECTS(!row_delta,
+                 "Not expecting window range to have value for string-based window calculation");
```

Comment:
This will never be reached because this function is only enabled if the tag is not a bounded one, and only bounded windows have non-null scalar pointer.